tock/hours/views.py

```diff
@@ -512,17 +512,52 @@ class ReportingPeriodListView(PermissionMixin, ListView):
     """ Currently the home view that lists the completed and missing time
     periods """
     context_object_name = "incomplete_reporting_periods"
-    queryset = ReportingPeriod.objects.all()
+    queryset = ReportingPeriod.objects.all().order_by('-start_date')
     template_name = "hours/reporting_period_list.html"
     permission_classes = (IsAuthenticated, )
 
+    def disallowed_dates(self, date):
+        # If the end of the fiscal year is not on a weekend, then create
+        # a list of dates around (the buffer) the end of the fiscal year
+        # where a reporting period will not be automatically created. This
+        # prevents the automatic creation of a reporting period that has
+        # working week days that span two fiscal years.
+        fy_start_date = datetime.date(year=date.year, month=10, day=1)
+        if fy_start_date.weekday() < 5 and date.month < 10:
+            buffer_days = 7 # A week before and after.
+            buffer_start = fy_start_date - datetime.timedelta(days=buffer_days)
+            buffer_end = fy_start_date + datetime.timedelta(days=buffer_days)
+            return [ buffer_start + datetime.timedelta(days=i) for i in \
+                range((buffer_end - buffer_start).days) ]
+        else:
+            return []
+
+    def auto_create_reporting_period(self):
+        # Automatically creates a new reporting period if the latest reporting
+        # period has concluded.
+        latest_rp = self.queryset.first()
+        if not latest_rp:
+            return None # In case there are no reporting periods created yet.
+        if latest_rp.end_date >= datetime.datetime.utcnow().date():
```

Comment:
i would pull out and only import the datetime functions you need in this file, i.e., `from datetime import datetime, timedelta, ...` but then you can avoid having to type datetime all over the place